app/models/comment.rb

```diff
@@ -1,4 +1,12 @@
 class Comment < ApplicationRecord
   belongs_to :author, class_name: 'User'
   belongs_to :post
+
+  after_save :update_comments_counter
+
+  def update_comments_counter
+    @posts.each do |post|
+      post.update_column(:CommentsCounter, post.comments.count) if post.id == post_id
+    end
+  end
```

Comment:
- [ ] Nice implementation of this logic you have here, however, instead of making use of the `@posts` which return all the Posts for all users as seen in the posts_controller #index where you defined the @posts instance variable, we can simply get the post and increment it's comment counter like this 👇 because a comment belongs to a post. This is more efficient

```ruby
def update_comments_counter
  post.increment!(:CommentsCounter)
end
```